torch/nested/_internal/ops.py

```diff
@@ -1471,6 +1471,58 @@ def slice_tensor(func, *args, **kwargs):
     return NestedTensor(func(inp._values, **new_kwargs), **extract_kwargs(inp))
 
 
+@register_jagged_func(
+    torch.ops.aten.index_put.default,
+    "input: jt_all, indices: any, values: t, accumulate: any?",
+)
+@register_jagged_func(
+    torch.ops.aten.index_put_.default,
+    "input: jt_all, indices: any, values: t, accumulate: any?",
+)
+def index_put_(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(  # type: ignore[misc]
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+
+    inp: NestedTensor = new_kwargs.pop("input")
+
+    # For index_put_ to work, we add together the indices of the ragged dimension
+    # and the batch dimension right before it, changing the batch indices by
+    # the offsets of each ragged dimension
+
+    indices = new_kwargs.pop("indices")
+
+    # Validate indices
+    if inp.lengths() is None:
+        lengths = inp.offsets().diff()
+    else:
+        lengths = inp.lengths()
+    torch._assert_async(
+        torch.all(indices[inp._ragged_idx] < lengths),
+        "Some indices in the ragged dimension are out of bounds!",
+    )
+
+    # Recompute indices for _values
+    ragged_indices = (
+        inp.offsets()[indices[inp._ragged_idx - 1]] + indices[inp._ragged_idx]
+    )
+    func_indices = (
```

Comment:
I think this is more correct:
```python
# Recompute indices for _values
ragged_indices = (
    inp.offsets()[indices[0]] + indices[inp._ragged_idx]
)
func_indices = (
    # before ragged dim
    indices[1: inp._ragged_idx]
    # ragged dim (combined with batch)
    + [ragged_indices]
    # after ragged dim
    + indices[inp._ragged_idx + 1 :]
)
```

with a small adjustment to take into account the case where ragged indices are not present in the inputs at all (e.g. for an empty `indices` tuple, an `indices` tuple that only contains a single element of batch indices, etc.)